misc/tracing-extensions/modules/ballerina-choreo-extension/src/main/java/org/ballerinalang/observe/trace/extension/choreo/ChoreoJaegerReporter.java

```diff
@@ -126,13 +126,15 @@ private synchronized void append(JaegerSpan jaegerSpan) {
             long duration = jaegerSpan.getDuration() / 1000;    // Jaeger stores duration in microseconds by default
             ChoreoTraceSpan traceSpan = new ChoreoTraceSpan(spanContext.getTraceId(), spanContext.getSpanId(),
                     jaegerSpan.getServiceName(), jaegerSpan.getOperationName(), timestamp, duration, tags, references);
-            traceSpans.add(traceSpan);
+            synchronized (traceSpans) {
+                traceSpans.add(traceSpan);
+            }
         }
 
         @Override
         public void run() {
             ChoreoTraceSpan[] spansToBeSent;
-            synchronized (this) {
+            synchronized (traceSpans) {
```

Comment:
what is the reason for this change? synchronizing on a different object.

Reply:
This change is not required. I reverted the change in 0b5ba74c9a423930d32f41451d597e6fc1dc2fe4